src/heap.cpp

```diff
@@ -12,32 +12,55 @@
 #include "search.h"
 
 
+CoreAnalyzerHeapInterface* gCAHeap;
+
 std::map<std::string, CoreAnalyzerHeapInterface*> gCoreAnalyzerHeaps;
 
-CoreAnalyzerHeapInterface* gCAHeap;
-void register_heap_managers() {
+static std::vector<void(*)()> gHeapInitFuncs = {
 	#ifdef WIN32
-	gCoreAnalyzerHeaps[HeapManagerMscrtMalloc] = get_mscrt_malloc_heap_manager();
-	gCAHeap = get_mscrt_malloc_heap_manager();
+    _init_mscrt_malloc,
 	#else
-	// heap.cpp is used by ref.dll, and we only support switching heap in Linux currently.
-	//gCoreAnalyzerHeaps.emplace(std::make_pair<std::string, CoreAnalyzerHeapInterface*>("pt", get_pt_malloc_heap_manager()));
-	//gCoreAnalyzerHeaps.emplace(std::make_pair<std::string, CoreAnalyzerHeapInterface*>("tc",get_tc_malloc_heap_manager()));
-	gCoreAnalyzerHeaps.emplace(std::make_pair<std::string, CoreAnalyzerHeapInterface*>("pt 2.35", get_pt_malloc_2_35_heap_manager()));
-	// default is pt malloc heap, in the future we will auto detect the heap.
-	gCAHeap = get_pt_malloc_2_35_heap_manager();
-	#endif
+    _init_pt_malloc_2_27,
+    _init_pt_malloc_2_31,
+    _init_pt_malloc_2_35,
+    //_init_tc_malloc,
+    #endif
+};
+
+bool init_heap_managers() {
+    gCoreAnalyzerHeaps.clear();
+    gCAHeap = nullptr;
+
+    for (auto f: gHeapInitFuncs)
+        f();
+
+    if (gCAHeap) {
+        CA_HEAP->init_heap();
```

Comment:
`init_heap` funcion looks similar as `gHeapInitFuncs`. Maybe it is more clear if we separate them with different names. From the name, someone might thinkg `init_heap` can be put inside `gHeapInitFuncs`

Reply:
Maybe something like `register_pt_malloca_2_27` or `add_pt_malloc_2_27` is more clear than `_init_pt_malloc_2_27`

Reply:
changed `gHeapInitFuncs` to `gHeapRegistrationFuncs`